app/src/forms/auth/middleware/userAccess.js

```diff
@@ -200,23 +204,29 @@ const hasSubmissionPermissions = (permissions) => {
 
       // Deleted submissions are inaccessible
       if (submissionForm.submission.deleted) {
-        return next(new Problem(401, { detail: 'You do not have access to this submission.' }));
+        throw new Problem(401, {
+          detail: 'You do not have access to this submission.',
+        });
       }
 
       // TODO: consider whether DRAFT submissions are restricted as deleted above
 
-      // Public (annonymous) forms are publicly viewable
+      // Public (anonymous) forms are publicly viewable
       const publicAllowed = submissionForm.form.identityProviders.find((p) => p.code === 'public') !== undefined;
       if (permissions.length === 1 && permissions.includes(Permissions.SUBMISSION_READ) && publicAllowed) {
         return next();
       }
 
       // check against the submission level permissions assigned to the user...
       const submissionPermission = await service.checkSubmissionPermission(req.currentUser, submissionId, permissions);
-      if (submissionPermission) return next();
+      if (!submissionPermission) {
+        // no access to this submission...
+        throw new Problem(401, {
+          detail: 'You do not have access to this submission.',
+        });
+      }
 
-      // no access to this submission...
-      return next(new Problem(401, { detail: 'You do not have access to this submission.' }));
+      return next();
```

Comment:
Just flipping the logic here so that the last thing that's done is call `next()`.